python/tests/tests.py

```diff
@@ -54,17 +54,39 @@ def tear_down_env(cls):
         cls.sql = None
 
 
+class TestTempDir(object):
+    @classmethod
+    def make_tempdir(cls):
+        cls.tempdir = tempfile.mkdtemp("sparkdl_tests", dir="/tmp")
+
+    @classmethod
+    def remove_tempdir(cls):
+        shutil.rmtree(cls.tempdir)
+
+
 class SparkDLTestCase(TestSparkContext, unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
         cls.setup_env()
-        cls.tempdir = tempfile.mkdtemp("sparkdl_tests", dir="/tmp")
+
 
     @classmethod
     def tearDownClass(cls):
         cls.tear_down_env()
-        shutil.rmtree(cls.tempdir)
 
     def assertDfHasCols(self, df, cols = []):
         map(lambda c: self.assertIn(c, df.columns), cols)
+
+
+class SparkDLTempDirTestCase(SparkDLTestCase, TestTempDir):
```

Comment:
I'd normally say we should just have the specific test case classes inherit from both... but since all our test case classes are subclasses of `SparkDLTestCase` we might as well have this...